homeassistant/components/kraken/sensor.py

```diff
@@ -138,80 +138,81 @@ def _handle_coordinator_update(self) -> None:
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "bid"
-                ][0]
-            if self._sensor_type == "bid_volume":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "bid"
-                ][1]
-            if self._sensor_type == "volume_today":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "volume"
-                ][0]
-            if self._sensor_type == "volume_last_24h":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "volume"
-                ][1]
-            if self._sensor_type == "volume_weighted_average_today":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "volume_weighted_average"
-                ][0]
-            if self._sensor_type == "volume_weighted_average_last_24h":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "volume_weighted_average"
-                ][1]
-            if self._sensor_type == "number_of_trades_today":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "number_of_trades"
-                ][0]
-            if self._sensor_type == "number_of_trades_last_24h":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "number_of_trades"
-                ][1]
-            if self._sensor_type == "low_today":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "low"
-                ][0]
-            if self._sensor_type == "low_last_24h":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "low"
-                ][1]
-            if self._sensor_type == "high_today":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "high"
-                ][0]
-            if self._sensor_type == "high_last_24h":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "high"
-                ][1]
-            if self._sensor_type == "opening_price_today":
-                self._state = self.coordinator.data[self.tracked_asset_pair_wsname][
-                    "opening_price"
-                ]
-            self._received_data_at_least_once = True  # Received data at least one time.
-        except TypeError:
-            if self._received_data_at_least_once:
-                if self._available:
-                    _LOGGER.warning(
-                        "Asset Pair %s is no longer available",
-                        self._device_name,
-                    )
-                    self._available = False
+        if self.coordinator.data:
```

Comment:
Invert this check and return early instead.

Reply:
Thanks! Done